### PR TITLE
Allow users to get direct file path on telegram bot

### PR DESCRIPTION
**Pull request by psyntium**
uses getFileLink method

### CODE REVIEW

File: telegrambot/99-telegrambot.js [hunk 1]
```diff
@@ -249,6 +249,12 @@ module.exports = function (RED) {
                                     msg.payload.path = path;
                                     node.send([msg, null]);
                                 });
+                            } else if (config.getFilePath && messageDetails.blob) {
+                                node.telegramBot.getFileLink(messageDetails.content).then(function(path) {
+                                    msg.payload.path = path;
+                                    node.send([msg, null]);
+                                });
+                            }
```
Comment (dvv) [inline telegrambot/99-telegrambot.js]: I believe this brace is a junk, no?

Comment (windkh) [reply telegrambot/99-telegrambot.js]: Well, the question is if the message should always contain the FileLink. Why should it be configurable?

Comment (psyntium) [reply telegrambot/99-telegrambot.js]: similar to the download.. getting the filelink would require another http call to the telegram bot.. so if it is not required, better not call it to speed up the request, no?

Comment (psyntium) [reply telegrambot/99-telegrambot.js]: @dvv what do you mean junk?

**Comment by psyntium** [comment]
Sounds good.. turning on by default may be a good idea since by right, if blob is true.. meaning theres an attachment that you may want to have access to..

**Comment by psyntium** [comment]
done as per suggested

**Comment by dvv** [comment]
Hmm. Now downloading depends hardly on getting weblink. If the latter fails -- no file is downloaded as well.
I believe these two are _separate, parallel, orthogonal_ branches.

**Comment by windkh** [comment]
I see what you mean and yes you are right. I will put it into separate steps that are executed after each other ...

**Comment by windkh** [comment]
Ok I came to the conclusion that I will not change it:
- if I change the order of the two async calls then I will probably have to duplicate a lot of code. 
- when getting the weblink fails, the download will probably fail, too
- vice versa when the download would fail you would not get a weblink either.
It's a liitle bit tricky, but right now I can't find a better solution

**Comment by dvv** [comment]
Well, according to https://github.com/yagop/node-telegram-bot-api/blob/master/src/telegram.js#L1408 and https://github.com/yagop/node-telegram-bot-api/blob/master/src/telegram.js#L1377 the bot client does almost what we try to do.
Let's just live with that.

**Pull request merged**